framework/src/media/utils/MediaUtils.cpp

```diff
@@ -1266,20 +1302,27 @@ float getSignalToNoiseRatio(const short *buffer, size_t size, int windows, int *
 	return energyRatio;
 }
 
-bool isMpeg2Ts(const unsigned char *buffer, size_t size)
+audio_type_t getAudioTypeFromStream(const unsigned char *buffer, size_t size)
 {
-	if (!buffer) {
-		meddbg("Invalid param, buffer is null!\n");
-		return false;
-	}
+	audio_type_t audioType;
+	unsigned int channel;
+	unsigned int sampleRate;
+	audio_format_type_t pcmFormat;
 
-	int ret = findTsPacket(buffer, size, true);
-	if (ret < UTILS_ERROR_NONE) {
-		meddbg("Not found TS packets, err %d\n", ret);
-		return false;
+	if (isMpeg2Ts(buffer, size)) {
+		audioType = AUDIO_TYPE_MP2T;
+	} else if (isMp3(buffer, size)) {
+		audioType = AUDIO_TYPE_MP3;
+	} else if (buffer_header_parsing(buffer, size, AUDIO_TYPE_AAC, &channel, &sampleRate, &pcmFormat)) {
```

Comment:
we don't need isXXX for aac & wave?

Reply:
buffer_header_parsing is enough to check aac & wave,
e.g. isXXX for aac like this:
```c
bool isAdts(const unsigned char *buffer, size_t size)
{
	unsigned int channel;
	unsigned int sampleRate;
	audio_format_type_t pcmFormat;
	return buffer_header_parsing(buffer, size, AUDIO_TYPE_AAC, &channel, &sampleRate, &pcmFormat);
}
```
is it necessary?

Reply:
umm... It seems to be more readable, I will add it.

Reply:
yes, that's what I mean. Thanks.